### PR TITLE
Added Linux Support

### PR DESCRIPTION
**Pull request by mysterywave**

### CODE REVIEW

**Review by eezstreet — APPROVED** [review]
Hi, submission approved. Though we will need to see about splitting Platform.cpp up into either multiple files or into different directories so as to not make it a monolithic file. But good job!

I should probably verify that this compiles on Windows as well, since I'm not sure if limits.h and uchar.h exist on Windows (I don't seem to see anything that suggests it doesn't, though)

File: CMakeLists.txt [hunk 1]
```diff
@@ -9,7 +9,12 @@ option(BUILD_D2CLIENT "Build D2Client" ON)
 option(BUILD_D2GAME "Build D2Game" ON)
 
 # Common options
-set(STATIC_LIBRARIES dbghelp.lib psapi.lib)
+if(WIN32)
+	set(STATIC_LIBRARIES dbghelp.lib psapi.lib)
+else()
+	set(CMAKE_CXX_STANDARD 11)
+	set(CMAKE_CXX_COMPILER g++)
```
Comment (Lectem) [inline CMakeLists.txt:16]: Aouch, please don't set the compiler from the cmakelists.txt...

File: CMakeLists.txt [hunk 2]
```diff
@@ -38,15 +43,23 @@ if(BUILD_GAME)
...
-	add_executable(game WIN32 ${GAME_SRC})
-	set_target_properties(game PROPERTIES OUTPUT_NAME game)
-	set_target_properties(game PROPERTIES LINKER_LANGUAGE CXX)
-
-	set(SDL_LIBRARY SDL2.lib SDL2main.lib SDL2test.lib SDL2_net.lib)
 
-	target_link_libraries(game ${STATIC_LIBRARIES} ${SDL_LIBRARY})
+	if(WIN32)
+		link_directories("Libraries/sdl/x86")
+		add_executable(game WIN32 ${GAME_SRC})
```
Comment (Lectem) [inline CMakeLists.txt:49]: why duplicate the stuff in common ? just make sure  ${SDL_LIBRARY} and  ${STATIC_LIBRARIES} are set correctly for all the platforms or just put the target_link_libraries in the if else

**Review by Lectem — COMMENTED** [review]
As @eezstreet said, I'm not sure moving from PlatformWindows.cpp to a common Platform.cpp is the right thing to do, a PlatformLinux.cpp would probably have been better.

**Pull request merged**